Server/mods/deathmatch/logic/luadefs/CLuaColShapeDefs.cpp

```diff
@@ -215,10 +215,12 @@ int CLuaColShapeDefs::CreateColRectangle ( lua_State* luaVM )
 int CLuaColShapeDefs::CreateColPolygon ( lua_State* luaVM )
 {
     //  colshape createColPolygon ( float fX, float fY, float fX1, float fY1, float fX2, float fY2, float fX3, float fY3, ... )
+
     std::vector < CVector2D > vecPointList;
 
+    // Get position and first 3 required points
     CScriptArgReader argStream ( luaVM );
-    for ( uint i = 0; i < 4 || argStream.NextCouldBeNumber (); i++ )
+    for ( uint i = 0; i < 4; i++ )
```

Comment:
Why didn't `(i < 4) || argStream.NextCouldBeNumber();` work before? It looks like it _should_ work (when clearly it didn't).

Did you test with `(i < 4) || argStream.NextIsVector2D();` first?

Reply:
The former did work fine with numbers only, it couldn't work with more than 3 points passed as vectors. The latter worked fine, I indeed tested it first. I guess I then decided to make Client and Server side code look more alike. 
I could of course do it by simply using that `for` loop (with fixed conditon) in Client side code too instead of `while` 😄 But I think it's just a cosmetic thing and the main difference is direction of code sync. Will change this if needed :)